stable/concourse/templates/secrets.yaml

```diff
@@ -9,31 +10,40 @@ metadata:
-  encryption-key: {{ default "" .Values.concourse.encryptionKey | b64enc | quote }}
-  old-encryption-key: {{ default "" .Values.concourse.oldEncryptionKey | b64enc | quote }}
-  {{ if .Values.credentialManager.vault }}
-  vault-ca-cert: {{ default "" .Values.credentialManager.vault.caCert | b64enc | quote }}
-  vault-client-token: {{ default "" .Values.credentialManager.vault.clientToken | b64enc | quote }}
-  vault-approle-id: {{ default "" .Values.credentialManager.vault.appRoleId | b64enc | quote }}
-  vault-approle-secret-id: {{ default "" .Values.credentialManager.vault.appRoleSecretId | b64enc | quote }}
-  vault-client-cert: {{ default "" .Values.credentialManager.vault.clientCert | b64enc | quote }}
-  vault-client-key: {{ default "" .Values.credentialManager.vault.clientKey | b64enc | quote }}
+  host-key: {{ .Values.secrets.concourse.hostKey | b64enc | quote }}
+  host-key-pub: {{ .Values.secrets.concourse.hostKeyPub | b64enc | quote }}
+  session-signing-key: {{ .Values.secrets.concourse.sessionSigningKey | b64enc | quote }}
+  worker-key: {{ .Values.secrets.concourse.workerKey | b64enc | quote }}
+  worker-key-pub: {{ .Values.secrets.concourse.workerKeyPub | b64enc | quote }}
+  {{ if not .Values.postgresql.enabled }}
+  postgresql-uri: {{ .Values.secrets.postgresql.uri | b64enc | quote }}
   {{ end }}
+  {{- if .Values.concourse.encryption.enabled }}
+  encryption-key: {{ default "" .Values.secrets.concourse.encryptionKey | b64enc | quote }}
+  old-encryption-key: {{ default "" .Values.secrets.concourse.oldEncryptionKey | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.concourse.basicAuth.enabled }}
+  basic-auth-username: {{ .Values.secrets.concourse.basicAuth.username | b64enc | quote }}
+  basic-auth-password: {{ .Values.secrets.concourse.basicAuth.password | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.concourse.githubAuth.enabled }}
+  github-auth-client-id: {{ default "" .Values.secrets.concourse.githubAuth.clientId | b64enc | quote }}
+  github-auth-client-secret: {{ default "" .Values.secrets.concourse.githubAuth.clientSecret | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.concourse.gitlabAuth.enabled }}
+  gitlab-auth-client-id: {{ default "" .Values.secrets.concourse.gitlabAuth.clientId | b64enc | quote }}
+  gitlab-auth-client-secret: {{ default "" .Values.secrets.concourse.gitlabAuth.clientSecret | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.concourse.genericOauth.enabled }}
+  generic-oauth-client-id: {{ default "" .Values.secrets.concourse.genericOauth.clientId | b64enc | quote }}
+  generic-oauth-client-secret: {{ default "" .Values.secrets.concourse.genericOauth.clientSecret | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.credentialManager.vault.enabled }}
+  vault-ca-cert: {{ default "" .Values.secrets.credentialManager.vault.caCert | b64enc | quote }}
+  vault-client-token: {{ default "" .Values.secrets.credentialManager.vault.clientToken | b64enc | quote }}
```

Comment:
I think it may be incorrect to persist the client token in this way as concourse will need to periodically renew it and redeploying a stale token would cause a failure. (Same for a pod failure or scaling) I haven't verified this so maybe Concourse is smart enough to deal with it and has the latest token stored elsewhere. Vault best practice is to either use an App Role or even better a service account with the Vault Kubernetes auth backend. That has the downside of longer lived credentials, but the advantage of being able to deal with long periods of downtime without requiring manually created tokens. I'm not sure if concourse has a way to save the refreshed tokens but a persistent volume might make more sense.

Reply:
>maybe Concourse is smart enough to deal with it and has the latest token stored elsewhere.